govcd/nsxt_edgegateway.go

```diff
@@ -755,6 +755,9 @@ func flattenEdgeGatewayUplinkToIpSlice(uplinks []types.EdgeGatewayUplinks) ([]ne
 // Special behavior:
 // * Passing nil minuend results in nil
 // * Passing nil subtrahend will return minuendSlice
+//
+// NOTE. This function will mutate minuendSlice to save memory and avoid having a copy of all values
+// which can become expensive if there are a lot of items
 func ipSliceDifference(minuendSlice, subtrahendSlice []netip.Addr) []netip.Addr {
```

Comment:
Just wondering whether this function should return nothing at all to state that is mutating `minuendSlice`?

Reply:
Apparently, I can't do exactly that (similar to why built-in `append` has similar mechanics).

https://forum.golangbridge.org/t/slice-pass-as-value-or-pointer/2866

> the contents of a slice argument can be modified by a function, but its header cannot. The length stored in the slice variable is not modified by the call to the function, since the function is passed a copy of the slice header, not the original. Thus if we want to write a function that modifies the header, we must return it as a result parameter, just as we have done here.


https://go.dev/blog/slices -> **Passing slices to functions**.

The way around it could be `*[]netip.Addr`, but this misses the semantics of `[]netip.Addr` everywhere.

Reply:
Very interesting read, thanks for sharing! So this is all good then 😄